lib/environment/dynamic_supervisor.ex

```diff
@@ -29,11 +30,13 @@ defmodule ApplicationRunner.Environment.DynamicSupervisor do
 
   @spec start_env(term()) ::
           {:error, {:already_started, pid()}} | {:ok, pid()} | {:error, term()}
-  defp start_env(env_metadata) do
+  defp start_env(%Environment.Metadata{} = env_metadata) do
     Logger.debug(
       "#{__MODULE__} Start Environment Supervisor with env_metadata: #{inspect(env_metadata)}"
     )
 
+    ApplicationServices.start_app(env_metadata.function_name)
```

Comment:
Is there a case where app_start fails to scale up but the call made in the tree startup succeeds in scaling up to 1?
If not, use a case to execute start_child only if we have scaled.

Reply:
Is the OpenFaaS app is scaled to zero, any call to it would scale it to one.

Reply:
Of course, but if for some reason we can't scale the application to scale 1 with start_app, the same reason will cause Envrionment to stop at startup.
This is just to avoid starting the environment supervisor if we know it's going to crash.

Reply:
I've handled it